woocommerce/woocommerce-ios
# [Orders with Coupons M7] Make coupons and discounts mutually exclusive

Closes: #10838 


## Description
This PR makes Coupons and Discounts mutually exclusive: 
* If no coupons have been added to an order, nor discounts to products, both are enabled.
* If a coupon has been added to an order, the discount row in products will be disabled.
* If a discount has been added to an order, the "Add coupon" row in the order will be disabled.

| Discount added -> Coupons disabled | None added -> Both enabled | Coupon added -> Discounts disabled  |
|--------|--------|--------|
| ![simulator_screenshot_1FA4A23F-3375-43E0-92D6-869C722B2162](https://github.com/woocommerce/woocommerce-ios/assets/3812076/bb7bea88-ab37-425e-a9f8-1cda9180f5f5) | ![simulator_screenshot_8FE221D0-B970-4AAA-8801-9F0186C8A7F0](https://github.com/woocommerce/woocommerce-ios/assets/3812076/6a61a611-6f6b-4d9b-84d4-a150daed259f) | ![simulator_screenshot_A49FD58F-A1AC-480D-84C4-2E012F73F613](https://github.com/woocommerce/woocommerce-ios/assets/3812076/eae3ef27-071d-4f36-9f0e-5f5a6db57b20) | 

## Changes
- Add a new `shouldDisallowDiscounts` property, which tracks if the order has any coupon.
- Modifies the existing `shouldDisableAddingCoupon` to keep track not only if the order is empty (so no coupons can be applied) but also if the "discounts total" is different than zero, in which case discounts have been applied.

## Testing instructions
- Switch `ordersWithCouponsM6` feature flag to `true`
- Add one or more products to an Order. Expand the product and observe how discounts can be added.
- Upon adding a discount, the `+ Add coupon` row will be disabled, while the tooltip icon is tappable.
- Removing the discount from the product will re-enable the coupons row.
- Adding a coupon to the order will disable the discounts row within each product, keeping the tooltip tappable.

![Simulator Screen Recording - iPhone 15 - 2023-10-11 at 17 27 32](https://github.com/woocommerce/woocommerce-ios/assets/3812076/86f4ef87-5cb5-4152-9bca-ffb8e4d3afc7)
